### ISSUE TITLE
G6 - Global nav menu-items are not being read out in Narrow view with NVDA 2015.4/Chrome (Develop Branch)

### ISSUE THREAD

**Issue by mmafe** [ccswbs/hjckrrh#273]
After updating to the newest version of NVDA -- NVDA 2015.4, I could no longer hear the global menu items on dev sites using the develop branch when listening via screen reader.

Occurs under the following conditions:

- Google Chrome
- NVDA 2015.4
- Global menu is open

On a dev site using the develop branch, e.g. https://aoda.web.uoguelph.ca/ccs/

1. Refresh page
2. Click the URL bar
3. Resize the page to narrow (responsive) width
4. Tab to the global navigation toggle
5. Open the global navigation toggle
6. Tab down through the global navigation menu

Do the same on the production site using the master branch, https://www.uoguelph.ca/ccs/

**Comment by mmafe** [comment]
@enichol and @jpinsent, could you please confirm whether or not you're seeing this issue on the previous version of NVDA in Chrome?

**Comment by mmafe** [comment]
It's also affecting my old version of Voiceover on my mac.

**Comment by mmafe** [comment]
@brharp Hey Brent, whenever you have a moment, could you check the following with JAWS?

On https://aoda.web.uoguelph.ca/ccs/

1. Refresh page
2. Resize the page to narrow (responsive) width
3. Tab to the global navigation toggle
4. Open the global navigation toggle
5. Tab down through the global navigation menu
6. Can you hear the global menu items?

**Comment by jpinsent** [comment]
Oh man, no idea how I missed this issue.

So far I've found that NVDA 2014.4 on Google Chrome will skip over both the expanded Global and Local menus in narrow view on the AODA website. When testing on live sites, the Local menu is read, but the Global menu is skipped.

The menus can occasionally be read by the screen reader, but only if you tab to it after opening the menu. Even then, sometimes I can get the screen reader to read the items in the menu, and other times it will stay silent even though my tab focus is on a menu item.

**Comment by mmafe** [comment]
No worries (we could have broken it since the last time you checked :) Plus we'll eventually have unit tests to check this kind of thing.)

It looks like the master branch isn't having issues with this, so I think we should just check out previous tags, and see when the issue first showed up. I'm going to do a quick run through now.

**Comment by brharp** [comment]
FWIW the global menu from https://aoda.web.uoguelph.ca/ccs/ is read properly by JAWS/IE

**Comment by mmafe** [comment]
Works in Firefox with latest NVDA, so I think the issue is definitely related to Chrome.

It may also be affecting VoiceOver -- I always see 2 types of focus: the keyboard focus gets stuck on the University of Guelph logo, while the visual focus proceeds through the menu, but does not announce the items.

We may be able to solve this since we need to change the toggle button so the next item is the menu anyway... https://github.com/ccswbs/hjckrrh/issues/125

**Comment by mmatthews** [comment]
I was able to recreate this issue in Chrome 47.0.2526.106 + NVDA 2015.4 on **production sites** for CCS and OAC.
To recreate, I followed the steps @mmafe shared above. It seemed to work properly at first but after expanding and collapsing the global nav a few times, NVDA would fail to read global nav items.

**Comment by mancusop** [comment]
Using Chrome/NVDA global & local nav menu in narrow view is read out loud after multiple expansions and collapses and it does not skip over the menu.

* My understanding is that this issue has been fixed with solution found in issue #329
